pandas/core/frame.py

```diff
@@ -494,6 +495,11 @@ def _init_dict(self, data, index, columns, dtype=None):
                 arrays.loc[missing] = [v] * missing.sum()
 
         else:
+            for key in data:
+                if isinstance(data[key], ABCIndexClass):
```

Comment:
really? we need to deep copy everything?

Reply:
I can be more specific and just make this `ABCDatetimeIndex`, but I don't see any solution that doesn't involve making a copy here.

The first thing I tried was to make a copy in `Block.setitem` (circa line 923 in internals.blocks) if we had a `DatetimeIndex`.  But that breaks a test because it means when we call `ser.__setitem__` (or `ser.loc.__setitem__`, or...), we fail to change other Series that are views on `ser`.

AFAICT the only way to do do this is to avoid ever having a `DatetimeIndex` and a `Series`/column share data.  This may be simpler to resolve after Blocks are backed by `DatetimeArray` instead of `DatetimeIndex` (I'm not 100% sure on this), but ATM this is a blocker to #24074, which is itself a blocker to `DatetimeArray`.  We could avoid having this be a blocker for #24074 if you were OK with the pickle workaround currently there.

Reply:
no, the pickle workaround is much worse. for now let's just do this for DTI w/tz only. otherwise this may be a lot of copying. Pls add a TODO around this though.